### ISSUE TITLE
Untransformed ErrorWithDiff.stack is passed to custom reporter

### ISSUE THREAD

**Issue by segrey** [vitest-dev/vitest#2195]
### Describe the bug

Hello, I'm working on a custom vitest reporter for IntelliJ/WebStorm integration.
Found that `Reporter.onFinished` is passed an untransformed (not mapped to source) `ErrorWithDiff.stack`:
For example, in [my test project](https://github.com/segrey/vitest/blob/main/examples/basic) assertion is failed on `…/test/suite.test.ts:9:…`, but `…/test/suite.test.ts:8:41` is passed to the reporter:
![image](https://user-images.githubusercontent.com/607109/197389766-9315065b-d232-4b8a-a04c-e2dbdd254b1b.png)

At the same time the correct stack is shown when running in the terminal:
![image](https://user-images.githubusercontent.com/607109/197389428-f09821ab-a6e1-4048-be4b-ac5277779cca.png)

As I can see `BaseReporter` calls [`ctx.logger.printError(error)`](https://github.com/vitest-dev/vitest/blob/4749e1f78f870121f585169f25c9f0b9a9274cfd/packages/vitest/src/node/reporters/base.ts#L320) which prints it to the console. Unfortunately, my reporter should communicate the stack to the IDE using another mechanism.

Is there a way to get ~untransformed~ transformed (mapped to source) `ErrorWithDiff.stack`? If there is no such way now, can it be added in future?

### Reproduction

Any failed test will reproduce the issue.
Here is what I used to demonstrate: https://github.com/segrey/vitest/blob/main/examples/basic/test/suite.test.ts#L9

### System Info

```shell
System:
    OS: Linux 5.4 Linux Mint 20.3 (Una)
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 14.51 GB / 31.21 GB
    Container: Yes
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 18.11.0 - ~/.nvm/versions/node/v18.11.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v18.11.0/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v18.11.0/bin/npm
  Browsers:
    Chrome: 106.0.5249.119
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by segrey** [comment]
Thank to the fix, in vitest >= 0.25.0, WebStorm/IntelliJ can show a failed assertion in the editor
![image](https://user-images.githubusercontent.com/607109/200511111-159a100c-bbe4-4307-87e4-0d3f4e6938a4.png)

@sheremet-va Nice work! :+1:

**Comment by DanielRamosAcosta** [comment]
Great 🤩 ! Thanks for the response @segrey
